### PR TITLE
feat: added Auth context and history context

### PR DESCRIPTION
**Pull request by Chakravarthi0**
added auth context, history context and protected routes

### CODE REVIEW

File: src/app-routes/AppRoutes.jsx [hunk 1]
```diff
@@ -1,19 +1,46 @@
...
       <Route path="/" element={<Home />}></Route>
       <Route path="/explore" element={<Explore />}></Route>
       <Route path="/video" element={<SingleVideo />}></Route>
-      <Route path="/signin" element={<SignIn/>}></Route>
-      <Route path="/signup" element={<SignUp/>}></Route>
-      <Route path="/playlist" element={<Playlist/>}></Route>
-      <Route path="/watchlater" element={<WatchLater/>}></Route>
-      <Route path="/history" element={<History/>}></Route>
-      <Route path="/single-playlist" element={<SinglePlaylist/>}></Route>
+      {!token ? (
```
Comment (shobit1337) [inline src/app-routes/AppRoutes.jsx:26]: You can create an AuthRoute. Similarly, you have implemented ProtectedRoutes. It will look cleaner.

**Review by shobit1337 — APPROVED** [review]
I'm not being able to sign in from your live link. Look into that.
Rest your code looks good to merge.

**Review by BelideNikhil — APPROVED** [review]
Clean code and nice UI
One suggestion is that guest login button font is not readable
Also you can hide the sidebar on login/signup page

**Comment by Chakravarthi0** [comment]
> I'm not being able to sign in from your live link. Look into that. Rest your code looks good to merge.

I forgot to add the env variable. Thanks for pointing it out

**Comment by Chakravarthi0** [comment]
> Clean code and nice UI One suggestion is that guest login button font is not readable Also you can hide the sidebar on login/signup page

Thanks for review, I will refactor the navbar and sidebar

**Pull request merged**